mathesar_ui/src/components/cell-fabric/data-types/boolean.ts

```diff
@@ -27,11 +29,11 @@ function getLabels(
 
 function getProps(
   column: BooleanLikeColumn,
-): SingleSelectCellExternalProps<boolean> {
+): SingleSelectCellExternalProps<boolean | null> {
   const labels = getLabels(column.display_options);
   return {
-    options: [true, false],
-    getLabel: (value?: boolean) => {
+    options: [null, true, false],
+    getLabel: (value?: boolean | null) => {
       if (isDefinedNonNullable(value)) {
         return value ? labels[0] : labels[1];
       }
```

Comment:
Do we need to allow `null` here? If we do, we should represent Null using the NULL component instead of empty string.

While we can allow null, there's no way for any other components (checkbox, text boxes, textarea etc.,) to ever set null as a value, so this would be a special option allowed only by the Select menu within a cell.

Reply:
> Do we need to allow `null` here?

The reason for this change is to improve the behavior of the input for boolean columns. Before this change, the input only had options for "true" and "false". The records selector needs to open with blank values for all inputs, and the user needs to be able to set a value and then set it _back_ to some sort of blank value after setting it. If you can think of a better way to accomplish what I'm after, then I'm open to changing the approach here. 

> we should represent Null using the NULL component instead of empty string.

I'm not sure about this. I think that would look pretty weird.

Reply:
>  I think that would look pretty weird.

I agree this would probably look weird in the context of record selector search. But we use the same component for the Cells in the table, and we represent nulls using the Null component in the Cell. The empty value looks odd in that context.

Either way, this is not a priority at the moment, and we can figure this out later.

Reply:
I just tested a boolean column. The dropdown shows an empty option. When you select the empty option the cell value changes to `NULL` and displays stylized. I think that should suffice. It would be nice to hide the empty option of the column doesn't allow NULL, but I don't imagine that's a priority.

Though it's worth noting that I did uncover #1919 while checking this out, and I predict that's a regression from this PR too. But I think it's worth merging this in and fixing that subsequently.